src/Infrastructure/Event/EventStreamProcessor.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Infrastructure\Event;
+
+use App\Domain\Event\EventBufferInterface;
+use App\Domain\Event\EventStreamProcessorInterface;
+use Broadway\Domain\DomainEventStream;
+use Broadway\EventHandling\EventBus;
+
+class EventStreamProcessor implements EventStreamProcessorInterface
```

Comment:
I really think this class should no exist and instead live in the event store. Example: https://github.com/broadway/broadway/pull/361/files#diff-ee264240ca5eabf48f566beb497573e0R91

Reply:
I'd prefer to have this logic in Broadway too. +1 for that PR!

Reply:
I'll rebuild this branch's logic on top of your Broadway PR.